### PR TITLE
Add `scala.quoted.Liftable`s to the stdlib

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: library/src/scala/quoted/Liftable.scala [hunk 1]
```diff
@@ -43,4 +45,195 @@ object Liftable {
...
+    def toExpr(iarray: IArray[T]): given QuoteContext => Expr[IArray[T]] = '{
+      val array = new Array[T](${iarray.length.toExpr})(ClassTag(${the[ClassTag[T]].runtimeClass.toExpr}))
+      ${ Expr.block(List.tabulate(iarray.length)(i => '{ array(${i.toExpr}) = ${iarray(i).toExpr} }), '{ array.asInstanceOf[IArray[T]] }) }
+    }
+  }
+
+  given [T: Type: Liftable] as Liftable[Seq[T]] = new Liftable[Seq[T]] {
+    def toExpr(seq: Seq[T]): given QuoteContext => Expr[Seq[T]] = given qctx => {
+      import qctx.tasty._
+      Repeated(seq.map(x => the[Liftable[T]].toExpr(x).unseal).toList, the[quoted.Type[T]].unseal).seal.asInstanceOf[Expr[Seq[T]]]
```
Comment (liufengyun) [inline library/src/scala/quoted/Liftable.scala]: Can we use `toExprOfSeq` here?

File: library/src/scala/quoted/Liftable.scala [hunk 2]
```diff
@@ -43,4 +45,195 @@ object Liftable {
...
+    }
+  }
+
+  given [T: Type: Liftable] as Liftable[List[T]] = new Liftable[List[T]] {
+    def toExpr(x: List[T]): given QuoteContext => Expr[List[T]] = x match {
+      case x :: xs  => '{ (${xs.toExpr}).::[T](${x.toExpr}) }
+      case Nil => '{ Nil: List[T] }
+    }
+  }
```
Comment (liufengyun) [inline library/src/scala/quoted/Liftable.scala]: can we use `toExprOfList` here?

File: library/src/scala/quoted/Liftable.scala [hunk 3]
```diff
@@ -43,4 +45,195 @@ object Liftable {
...
+    def toExpr(x: Either[L, R]): given QuoteContext => Expr[Either[L, R]] = x match {
+      case Left(x) => '{ Left[L, R](${x.toExpr}) }
+      case Right(x) => '{ Right[L, R](${x.toExpr}) }
+    }
+  }
+
+  given [T1: Type: Liftable] as Liftable[Tuple1[T1]] = new {
+    def toExpr(tup: Tuple1[T1]) =
+      '{ Tuple1(${tup._1.toExpr}) }
+  }
```
Comment (liufengyun) [inline library/src/scala/quoted/Liftable.scala]: Minor: maybe still write `new Liftable[...]` to be consistent and explicit as above, the same for the code below

File: library/src/scala/quoted/Liftable.scala (hunk 1 above)
Comment (nicolasstucki) [reply library/src/scala/quoted/Liftable.scala]: Yes, now I can rename and use it

File: library/src/scala/quoted/Liftable.scala (hunk 2 above)
Comment (nicolasstucki) [reply library/src/scala/quoted/Liftable.scala]: Yes, we can

File: library/src/scala/quoted/Liftable.scala (hunk 3 above)
Comment (nicolasstucki) [reply library/src/scala/quoted/Liftable.scala]: It got way too long and made it harder to read.

Comment (nicolasstucki) [reply library/src/scala/quoted/Liftable.scala]: Specially for Tuple22

**Pull request merged**